extensions/npc/standalone.mjs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+import fs from 'fs';
+import InteractiveObjectServerBaseClass from '../../common/interactive-object/server.mjs';
+
+/**
+ * TODO: jsdoc
+ */
+class SingleNPC extends InteractiveObjectServerBaseClass {}
+
+/**
+ * TODO: jsdoc
+ */
+class Standalone {
+  /**
+   * Create the standalone extension service object but does not start it.
+   * @constructor
+   * @param {ExtensionHelper} helper - An extension helper object for servicing
+   * various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+    this.NPCs = new Map();
+
+    // create individual NPC
+    fs.readdirSync('../run/npc').forEach((file) => {
+      const npcName = file.slice(0, -('.json'.length));
+      const npc = new SingleNPC(helper, npcName, `../run/npc/${file}`);
+      this.NPCs.set(npcName, npc);
+    });
+  }
+
+  /**
+   * Initializes the extension.
+   */
+  async initialize() {
+  }
+
+  /**
+   * Return the ejs partials for the client part of this extension.
+   * @return {object} partials - An object, it could have the following:
+   * inDiv - A string to the path of ejs partial for the location inDiv.
+   */
+  static getPartials() {
+    return {inDiv: 'in-div.ejs'};
+  }
+
+  /**
+   * TODO
+   * @param {Object} player - TODO
+   * @param {String} npcName - TODO
+   * @return {Object}
+   */
+  async c2s_getDisplayInfo(player, npcName) {
+    const npc = this.NPCs.get(npcName);
+    if (typeof npc === 'undefined') return {};
+    return npc.getDisplayInfo();
+  }
+
+  /**
+   * TODO
+   * @param {Object} player - TODO
+   * @param {String} npcName - TODO
+   * @return {mapCoord}
+   */
+  async c2s_getInitialPosition(player, npcName) {
+    const npc = this.NPCs.get(npcName);
+    if (typeof npc === 'undefined') return null;
+    return npc.getInitialPosition();
+  }
+
+  /**
+   * TODO
+   * @param {Object} player - TODO
+   * @param {String} npcName - TODO
+   */
+  async c2s_startInteraction(player, npcName) {
+    const npc = this.NPCs.get(npcName);
+    if (typeof npc === 'undefined') return;
```

Comment:
Optional: Return failure and/or log an error so it doesn't fail silently? Ditto above.